vallie/vallie.go

```diff
@@ -0,0 +1,30 @@
+// Package vallie stands in for a stand-alone validation library
+// purely for discussion purposes.
+//
+// Assuming this is a desired approach, more fidelity will be added
+// and vallie would be extracted into it's own repo
+package vallie
+
+import (
+	"reflect"
+
```

Comment:
Yeah the name is a total placeholder. The reason we probably wouldn't want to keep it as part of dig is because this validation library could be very useful without any connection to dig. It's very common to check non-nils for pointers and interfaces when you write an exported function. It's similar to checking for nulls in java and the thought was to provoke a library capable of solving both problems, even if it's primary goal is to serve dig.